src/gluonts/model/forecast_generator.py

```diff
@@ -116,17 +115,21 @@ def __call__(
     ) -> Iterator[Forecast]:
         for batch in inference_data_loader:
             inputs = select(input_names, batch, ignore_missing=True)
-            outputs = predict_to_numpy(prediction_net, inputs)
-            if output_transform is not None:
-                outputs = output_transform(batch, outputs)
+            (outputs,), loc, scale = prediction_net(*inputs.values())
+            if scale is not None:
+                outputs = outputs * scale[..., None]
+            if loc is not None:
+                outputs = outputs + loc[..., None]
 
+            if output_transform is not None:
+                log_once(OUTPUT_TRANSFORM_NOT_SUPPORTED_MSG)
             if num_samples:
                 log_once(NOT_SAMPLE_BASED_MSG)
 
             i = -1
-            for i, output in enumerate(outputs):
+            for i, output in enumerate(to_numpy(outputs)):
                 yield QuantileForecast(
-                    output,
+                    output.T,
```

Comment:
QuantileForecast expects array of shape `[num_quantiles, prediction_length]`, but all models usually produce the output of shape `[batch_size, prediction_length, *additional_dims]` - both in case of `DistributionOutput` and `QuantileOutput`. I think it's better to keep model output shape consistent and do the transpose here.